session/pessimistic_test.go

```diff
@@ -391,3 +393,30 @@ func (s *testPessimisticSuite) TestOptimisticConflicts(c *C) {
 	tk.MustExec("commit")
 	tk.MustQuery("select * from conflict").Check(testkit.Rows("1 3"))
 }
+
+func (s *testPessimisticSuite) TestWaitLockKill(c *C) {
+	// Test kill command works on waiting pessimistic lock.
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk2 := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists test_kill")
+	tk.MustExec("create table test_kill (id int primary key, c int)")
+	tk.MustExec("insert test_kill values (1, 1)")
+	tk.MustExec("begin pessimistic")
+	tk2.MustExec("begin pessimistic")
+	tk.MustQuery("select * from test_kill where id = 1 for update")
```

Comment:
How about adding a test case for `MAX_EXECUTION_TIME`?

Reply:
I test it manually and currently `MAX_EXECUTION_TIME` works when the session is waiting for a lock.
Does MySQL have different behavior? @jackysp

Reply:
MySQL has the same behavior.